cookbook-recipes.md

```diff
@@ -0,0 +1,659 @@
+
+`npm install axios uuid jsonwebtoken node-forge`
+
+### Table of Contents
+
+- [Common constants and functions](#common)
+- [Consumer makes call without pre-authorization](#client-without-preauth)
+- [Consumer makes call with pre-authorization](#client-with-preauth)
+- [Provider is called without pre-authorization](#provider-without-preauth)
+- [Provider is called with pre-authorization](#provider-with-preauth)
+- [Provider is called without external authorization](#provider-without-auth)
+
+<a name="common"></a>
+
+### Common constants and functions
+
+All recipes make use of a number of constants and functions defined below. Some constants should be adapted by you, such as the location of your private key and certificate.
+
+```
+// requires
+const jwt = require('jsonwebtoken');
+const { v4: uuidv4 } = require('uuid');
+const fs = require('fs');
+const crypto = require('crypto');
+const axios = require('axios');
+const forge = require('node-forge');
+
+// file paths
+
+const privateKeyPath = process.env.HOME + '/.ssh/EU.EORI.NLFLEXTRANS.pem'; // NOTE: Example definition, adjust as needed
+const certKeyPath = process.env.HOME + '/.ssh/EU.EORI.NLFLEXTRANS.crt'; // NOTE: Example definition, adjust as needed
+
+// constants
+
+const YOUR_EORI = "EU.EORI.NLFLEXTRANS"; // NOTE: Example definition, adjust as needed
+const ASSOC_EORI = "EU.EORI.NLDILSATTEST1"; // NOTE: Example definition, adjust as needed
+const SP_EORI = "EU.EORI.NL809023854"; // NOTE: Example definition, adjust as needed
+const AR_EORI = 'EU.EORI.NL000000004'; // NOTE: Example definition, adjust as needed
+
+// credentials
+
+const pemData = fs.readFileSync(privateKeyPath, 'utf8');
+const publicKey = crypto.createPublicKey(pemData);
+const certificateChainData = fs.readFileSync(certKeyPath, 'utf8');
+// Split the certificate chain into individual certificates
+const certificates = certificateChainData.match(/-----BEGIN CERTIFICATE-----[\s\S]+?-----END CERTIFICATE-----/g);
+// Convert each certificate to DER format and then base64 encode it
+const x5c = certificates.map(cert => {
+  return cert.replace(/-----\w+ CERTIFICATE-----/g, '').replace(/\s+/g, '');
+});
+
+let tokenList = {};
+
+// URLs
+
+const assocUrlRoot = "https://dilsat1-mw.pg.bdinetwork.org"; // NOTE: Example definition, adjust as needed
+const tokenUrlAssoc = assocUrlRoot + "/connect/token";
+const partiesUrlAssoc = assocUrlRoot + "/parties";
+const trustedUrlAssoc = assocUrlRoot + "/trusted_list";
+// const tokenArUrl = "https://ar.isharetest.net/connect/token"; // NOTE: Example definition, adjust as needed
+// const delegationArUrl = "https://ar.isharetest.net/delegation"; // NOTE: Example definition, adjust as needed
+
+// create client assertion with default values
+function createClientAssertion(token) {
+  return new URLSearchParams({
+    "grant_type": "client_credentials",
+    "scope": "iSHARE",
+    "client_id": YOUR_EORI,
```

Comment:
Je gebruikt  `YOUR_EORI` voor de server id en voor client id. Dat is erg verwarrend.